### PR TITLE
Fixed an issue where folder provider icons would not show

### LINKED ISSUE #5274: [v9.11.0-rc4] Folder icon for custom folder provider fails to load
## Description of bug
Folder icon for custom folder provider fails to load (404) within the new Resource Manager module. The issue appears to be due to the Resource Manager module requesting the image with .svg extension, when it is defined in the folder provider code (and installed on disk) as .png.

## Steps to reproduce
1. An existing custom folder provider is installed which defines the folder icon path as ~/Providers/FolderProviders/TfsFolderProvider/icon-asset-manager-tfs-folder.png (see code image below)
2. After upgrade to v9.11.0-rc4, within the new Resource Manager module, the folder icon image is broken (see image below)

## Expected behavior
The folder icon should display as it did before the v9.11.0-rc4 upgrade. No change to folder provider code should be required.

## Screenshots
image
image
image

## Affected version
[ X ] 09.11.00 release candidate

### PR DESCRIPTION
**Pull request by valadas**
Closes #5274

The new logic is as follows:
1. If we don't have any folder mapping info, we show the standard folder svg
2. If we have in the resource manager an svg for the provider, we use that one (usually nicer and we have one for our own Azure profider connector)
3. If the provider has one, we use the one the provider dictates If all else fails, we use the standard svg one.

In addition to this, I also added a button so users can create new folder mappings (totally forgot about that part when implementing the feature).

### CODE REVIEW

**Review by david-poindexter — APPROVED** [review]
Well...that looks like it was fun!  🎉 🌮

**Review by mitchelsellers — APPROVED** [review]

**Review by bdukes — APPROVED** [review]
Logic looks good, thanks!

File: DNN Platform/Modules/ResourceManager/ResourceManager.Web/src/components/dnn-rm-folder-mappings/dnn-rm-folder-mappings.tsx [hunk 1]
```diff
@@ -53,6 +66,23 @@ export class DnnRmFolderMappings {
                         </tbody>
                     </table>
                 }
+                {this.addFolderTypeUrl &&
+                    <div class="controls">
+                        <dnn-button
+                            type='primary'
```
[the remark this answers is not in the document]
Comment (valadas) [reply DNN Platform/Modules/ResourceManager/ResourceManager.Web/src/components/dnn-rm-folder-mappings/dnn-rm-folder-mappings.tsx:72]: Yes, the cancel one is reversed like so. Here primary, secondary, tertiary sets the color and you can use `reversed`, `disabled`, `size`, etc to further configure the button.

It renders as such in this case:

image

That being said, `primary` is the default, so let me improve that and remove the type...

File: DNN Platform/Modules/ResourceManager/Services/ItemsController.cs [hunk 2]
```diff
@@ -670,14 +670,28 @@ private static string GetFileIconUrl(string extension)
...
+            if (File.Exists(HttpContext.Current.Server.MapPath($"~/{localSvg}")))
+            {
+                return Globals.ApplicationPath + "/" + localSvg;
+            }
+
+            if (File.Exists(HttpContext.Current.Server.MapPath(folderMapping.ImageUrl)))
+            {
+                var path = folderMapping.ImageUrl.Replace("~", HttpContext.Current.Request.ApplicationPath)
+                    .TrimStart('/');
+                return $"/{path}";
```
[the remark this answers is not in the document]
Comment (valadas) [reply DNN Platform/Modules/ResourceManager/Services/ItemsController.cs:691]: That one got me scratching my head a bit there. Open source is the best school! So much cleaner

[the remark this answers is not in the document]
Comment (valadas) [reply DNN Platform/Modules/ResourceManager/Services/ItemsController.cs:691]: Yeah, that one does not work in context, it returns a ~/.... type path

image

but I'll play what other stuff in this namespace to see what's what...

[the remark this answers is not in the document]
Comment (valadas) [reply DNN Platform/Modules/ResourceManager/Services/ItemsController.cs]: ToAbsolute appears to do it in this context... See my last commit which should handle all the requested changes

**Comment by valadas** [comment]
Last commit should handle all the provided feedback and appears to work for all scenarios:

image

File: DNN Platform/Modules/ResourceManager/Services/ItemsController.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (bdukes) [reply DNN Platform/Modules/ResourceManager/Services/ItemsController.cs]: Ah, yeah, sorry for pointing you in the wrong direction. It looks like [you can maybe use `Url.Content` to resolve the tilde](https://stackoverflow.com/a/33404533/2688) (from [`ApiController.Url`](https://docs.microsoft.com/en-us/previous-versions/aspnet/mt134840(v=vs.118)) to [`UrlHelper.Content`](https://docs.microsoft.com/en-us/previous-versions/aspnet/mt153121(v=vs.118)))

**Review by bdukes — APPROVED** [review]
Great, thanks!

File: DNN Platform/Modules/ResourceManager/Services/ItemsController.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (valadas) [reply DNN Platform/Modules/ResourceManager/Services/ItemsController.cs]: Would you like me to try/test this suggestion or what I did with VirtualPathUtility in my last commit is good to go ?

[the remark this answers is not in the document]
Comment (bdukes) [reply DNN Platform/Modules/ResourceManager/Services/ItemsController.cs]: What you have is good, no worries.

[the remark this answers is not in the document]
Comment (valadas) [reply DNN Platform/Modules/ResourceManager/Services/ItemsController.cs]: Cool, can you merge or did I miss anything pending ?

**Pull request merged**